### PR TITLE
Add Documentation for R4 Organization Read and Searches

### PR DESCRIPTION
**Pull request by sbanda51**
Adding documentation for R4 Organization Read and Search.

### CODE REVIEW

File: lib/resources/r4/allergy_intolerance.yaml [hunk 1]
```diff
@@ -77,7 +77,7 @@ fields:
     description: The clinical status of the allergy or intolerance.
     terminology:
     - display: AllergyIntolerance Clinical Status Codes
-      system: http://terminology.hl7.org/CodeSystem/allergyintolerance-clinical
+      system: http://terminology.hl7.org/CodeSystem/aller gyintolerance-clinical
```
Comment (nishanthnamana) [inline lib/resources/r4/allergy_intolerance.yaml]: nit: Spelling

Comment (sbanda51) [reply lib/resources/r4/allergy_intolerance.yaml]: my bad reverted the change https://github.com/cerner/fhir.cerner.com/pull/325/commits/96dd8b454f4f1f5d3b5ef538fbbafdb623da089d

File: lib/resources/r4/organization.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    cardinality: 0..*
+    type: Identifier
+    description: The identifier for the Organization.
+    children:
+
+      - name: type
+        required: 'No'
+        cardinality: 0..1
+        type: CodeableConcept
+        description: The type for the identifier that can be used to determine which identifier to use for a specific purpose.
```
Comment (ebaranowski) [inline lib/resources/r4/organization.yaml]: A coded type for the identifier that can be used to determine which identifier to use for a specific purpose.

File: lib/resources/r4/organization.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        url: http://hl7.org/fhir/r4/datatypes-definitions.html#Identifier.type
+        binding:
+          description: The type used to determine which identifier to use for a specific purpose.
```
Comment (ebaranowski) [inline lib/resources/r4/organization.yaml]: A coded type for the identifier that can be used to determine which identifier to use for a specific purpose.

File: lib/resources/r4/organization.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    required: 'No'
+    cardinality: 0..*
+    type: CodeableConcept
+    description: The type for the Organization.
+    binding:
+      description: Indicates the type of Organization.
+      terminology:
+        - display: OrganizationType
+          system: http://hl7.org/fhir/ValueSet/organization-type
+          info_link: http://terminology.hl7.org/CodeSystem/organization-type.html
```
Comment (ebaranowski) [inline lib/resources/r4/organization.yaml]: http://hl7.org/fhir/r4/valueset-organization-type.html

File: lib/resources/r4/organization.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  - name: type
+    required: 'No'
+    cardinality: 0..*
+    type: CodeableConcept
+    description: The type for the Organization.
+    binding:
+      description: Indicates the type of Organization.
+      terminology:
+        - display: OrganizationType
+          system: http://hl7.org/fhir/ValueSet/organization-type
```
Comment (ebaranowski) [inline lib/resources/r4/organization.yaml]: If this is mapped to this system then we mapped it wrong. It should be mapped to this system:

http://terminology.hl7.org/CodeSystem/organization-type

File: lib/resources/r4/organization.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        cardinality: 0..1
+        type: CodeableConcept
+        description: The type for the identifier that can be used to determine which identifier to use for a specific purpose.
+        url: http://hl7.org/fhir/r4/datatypes-definitions.html#Identifier.type
+        binding:
+          description: The type used to determine which identifier to use for a specific purpose.
+          terminology:
+            - display: Identifier Type Codes
+              system: http://hl7.org/fhir/identifier-type
+              info_link: http://terminology.hl7.org/CodeSystem/v2-0203.html
```
Comment (ebaranowski) [inline lib/resources/r4/organization.yaml]: http://hl7.org/fhir/r4/valueset-identifier-type.html

File: lib/resources/r4/organization.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            - display: v2 Identifier Type
+              system: http://terminology.hl7.org/CodeSystem/v2-0203
+              info_link: https://hl7.org/fhir/R4/v2/0203/index.html
```
Comment (ebaranowski) [inline lib/resources/r4/organization.yaml]: http://hl7.org/fhir/r4/valueset-identifier-type.html

File: lib/resources/r4/organization.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        required: 'No'
+        cardinality: 0..1
+        type: CodeableConcept
+        description: The type for the identifier that can be used to determine which identifier to use for a specific purpose.
+        url: http://hl7.org/fhir/r4/datatypes-definitions.html#Identifier.type
+        binding:
+          description: The type used to determine which identifier to use for a specific purpose.
+          terminology:
+            - display: Identifier Type Codes
+              system: http://hl7.org/fhir/identifier-type
```
Comment (ebaranowski) [inline lib/resources/r4/organization.yaml]: I dont see this system.... Did we map it wrong again?

File: content/millennium/r4/proprietary-codes-and-systems.md [hunk 9]
```diff
@@ -61,6 +61,17 @@ The Encounter Resource supports proprietary codes for:
...
+
+<!-- use html header to avoid showing up in toc -->
+<h4>Organization</h4>
+
+The Organization Resource supports proprietary codes for:
+
+* Organization.identifier.type codes are maintained in [Code Set 334 Organization Alias Type](#code-set-334-organization-alias-type)
+* Organization.type codes are maintained in [Code Set 278 Organization Type](#code-set-278-organization-type)
+
+
```
[the remark this answers is not in the document]
Comment (sbanda51) [reply content/millennium/r4/proprietary-codes-and-systems.md]: fixed https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-6fdcb00b5b1b6edd51b687794bea6b95L74

File: content/millennium/r4/proprietary-codes-and-systems.md [hunk 10]
```diff
@@ -287,6 +298,17 @@ This code set maintains medication administration sites such as left arm, right
       "display": "Right arm", 
       "userSelected": true 
     }
+    
+##### Code Set 278 Organization Type
+
+This code set maintains Organization Types such as GOVT, INSURANCE
```
[the remark this answers is not in the document]
Comment (sbanda51) [reply content/millennium/r4/proprietary-codes-and-systems.md]: done https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-6fdcb00b5b1b6edd51b687794bea6b95R303

File: content/millennium/r4/proprietary-codes-and-systems.md [hunk 11]
```diff
@@ -309,6 +331,17 @@ This code set maintains the various personnel positions in the facility such as
       "display": "Attending Physician",
       "userSelected": true
     }
+    
+##### Code Set 334 Organization Alias Type
+
+This code set maintains the Organization Alias Types and currently only supports NPI, OID, FederalTAXID.
```
[the remark this answers is not in the document]
Comment (sbanda51) [reply content/millennium/r4/proprietary-codes-and-systems.md]: done https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-6fdcb00b5b1b6edd51b687794bea6b95R336

File: lib/resources/example_json/r4_examples_organization.rb [hunk 12]
```diff
@@ -0,0 +1,232 @@
+module Cerner
+  module Resources
+ R4_ORGANIZATION_ENTRY ||= {
```
[the remark this answers is not in the document]
Comment (sbanda51) [reply lib/resources/example_json/r4_examples_organization.rb]: fixed https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-868f68eb9a3f0c1d02dd56334fdb25b3L3

File: lib/resources/r4/organization.yaml [hunk 13]
```diff
@@ -0,0 +1,44 @@
+---
+name: Organization
+field_name_base_url: http://hl7.org/fhir/r4/organization-definitions.html#Organization
+fields:
+  - name: identifier
+    required: 'No'
+    cardinality: 0..*
+    type: Identifier
+    description: The identifier for the Organization.
```
[the remark this answers is not in the document]
Comment (sbanda51) [reply lib/resources/r4/organization.yaml]: done https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-b08823bfbc91ea717475d1d2372baf64R9

File: lib/resources/r4/organization.yaml (hunk 2 above)
Comment (sbanda51) [reply lib/resources/r4/organization.yaml]: done https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-b08823bfbc91ea717475d1d2372baf64R19

File: lib/resources/r4/organization.yaml (hunk 3 above)
Comment (sbanda51) [reply lib/resources/r4/organization.yaml]: done https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-b08823bfbc91ea717475d1d2372baf64R19

File: lib/resources/r4/organization.yaml (hunk 4 above)
Comment (sbanda51) [reply lib/resources/r4/organization.yaml]: updated https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-b08823bfbc91ea717475d1d2372baf64R23

File: lib/resources/r4/organization.yaml (hunk 5 above)
Comment (sbanda51) [reply lib/resources/r4/organization.yaml]: my updated https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-b08823bfbc91ea717475d1d2372baf64R37

File: lib/resources/r4/organization.yaml (hunk 6 above)
Comment (sbanda51) [reply lib/resources/r4/organization.yaml]: https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-b08823bfbc91ea717475d1d2372baf64R37

File: lib/resources/r4/organization.yaml (hunk 7 above)
Comment (sbanda51) [reply lib/resources/r4/organization.yaml]: https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-b08823bfbc91ea717475d1d2372baf64R23

File: lib/resources/r4/organization.yaml (hunk 8 above)
Comment (sbanda51) [reply lib/resources/r4/organization.yaml]: removed here https://github.com/cerner/fhir.cerner.com/pull/325/commits/cb6f3d36fa7afcdabc3306d00cfa78017b470b11#diff-b08823bfbc91ea717475d1d2372baf64L22 as it was copy paste error from patient resource.

**Comment by ebaranowski** [comment]
Need to put this if statement: https://github.com/cerner/fhir.cerner.com/blob/organization_read_searches/layouts/encounters.html#L5

around:

https://github.com/cerner/fhir.cerner.com/blob/organization_read_searches/layouts/entities.html#L7

**Comment by ebaranowski** [comment]
Validation:

Screen Shot 2019-12-03 at 1 32 14 PM
Screen Shot 2019-12-03 at 1 32 59 PM
Screen Shot 2019-12-03 at 1 33 06 PM
Screen Shot 2019-12-03 at 1 33 30 PM
Screen Shot 2019-12-03 at 1 33 30 PM
Screen Shot 2019-12-03 at 1 33 37 PM

**Comment by ebaranowski** [comment]
Additional Validation:

Screen Shot 2019-12-03 at 1 40 28 PM

**Comment by sbanda51** [comment]
@ebaranowski Added the changes requested https://github.com/cerner/fhir.cerner.com/pull/325/commits/bbe548e58af25f42555b3a0bdf04f0c686d29c0c

File: content/millennium/r4/entities/organization.md [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+* [Contact Information]( https://hl7.org/fhir/r4/organization-definitions.html#Organization.telecom){:target="_blank"}
+* [Address]( https://hl7.org/fhir/r4/organization-definitions.html#Organization.address){:target="_blank"}
+
+## Terminology Bindings
+
+<%= terminology_table(:organization, :r4) %>
+
+## Search
+
+Search for Organization that meet supplied query parameters:
```
Comment (nishanthnamana) [inline content/millennium/r4/entities/organization.md]: nit: Organizations

File: content/millennium/r4/entities/organization.md [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+
+#### Request
+
+    GET https://fhir-open.sandboxcerner.com/r4/0b8a0111-e8e6-4c26-a91c-5069cbc6b1ca/Organization?_id=1750115
+
+#### Response
+
+<%= headers status: 200 %>
+<%= json(:r4_organization_bundle) %>
+
```
Comment (nishanthnamana) [inline content/millennium/r4/entities/organization.md]: Its better to declaimer here saying the same result is not guaranteed. You can use the below tag
``<%= disclaimer %>``

Applies for all the examples.

Comment (sbanda51) [reply content/millennium/r4/entities/organization.md]: Added https://github.com/cerner/fhir.cerner.com/pull/325/commits/618e85c15cc0f3e71f36acb0867eb9e24789c3dc

File: content/millennium/r4/entities/organization.md (hunk 14 above)
Comment (sbanda51) [reply content/millennium/r4/entities/organization.md]: updated https://github.com/cerner/fhir.cerner.com/pull/325/commits/618e85c15cc0f3e71f36acb0867eb9e24789c3dc

File: lib/resources/r4/organization.yaml [hunk 16]
```diff
@@ -0,0 +1,41 @@
+---
+name: Organization
+field_name_base_url: http://hl7.org/fhir/r4/organization-definitions.html#Organization
```
Comment (nishanthnamana) [inline lib/resources/r4/organization.yaml]: All the links in this file, should be secured.

File: content/millennium/r4/entities/organization.md (hunk 14 above)
Comment (nishanthnamana) [reply content/millennium/r4/entities/organization.md]: +1

File: content/millennium/r4/entities/organization.md (hunk 15 above)
Comment (nishanthnamana) [reply content/millennium/r4/entities/organization.md]: +1

File: lib/resources/r4/organization.yaml (hunk 16 above)
Comment (sbanda51) [reply lib/resources/r4/organization.yaml]: done https://github.com/cerner/fhir.cerner.com/pull/325/commits/981f8f421bb02f636415ef251b731508616a52d6

**Comment by nishanthnamana** [comment]
fhir.cerner.com Verification

Organization_1

Organization_2


Organization_3

**Pull request merged**